Core/Src/stm32f4xx_it.c

```diff
@@ -183,6 +185,22 @@ void SysTick_Handler(void)
 /* please refer to the startup file (startup_stm32f4xx.s).                    */
 /******************************************************************************/
 
+/**
+  * @brief This function handles DMA1 stream3 global interrupt.
+  */
+void DMA1_Stream3_IRQHandler(void)
+{
+  /* USER CODE BEGIN DMA1_Stream3_IRQn 0 */
+
+  HAL_UART_TxCpltCallback(&huart3); //jake

```

Comment:
According to the documentation, "At transmission end of [UART DMA] transfer, HAL_UART_TxCpltCallback is executed", so this should not be in the DMA IRQ handler becuase it will be called anway. Also get rid of the IRQ handler because everything is taken care of by HAL_UART_TxCpltCallback. I think I told you to do both of these things so my bad.